{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "#need to pip install google-cloud-batch first time you run\n",
    "import pandas as pd\n",
    "import os\n",
    "import shutil\n",
    "import numpy as np\n",
    "import subprocess\n",
    "import glob\n",
    "import sys\n",
    "import gzip\n",
    "from google.cloud import batch_v1\n",
    "\n",
    "# Supress copy warning.\n",
    "pd.options.mode.chained_assignment = None"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "{\"time\":\"07/03/2025 02:04:03.442142\",\"severity\":\"INFO\",\"msg\":\"Start gcsfuse/1.2.0 (Go version go1.21.0) for app \\\"\\\" using mount point: /home/levineks/gp2_genotools_data\\n\"}\n"
     ]
    }
   ],
   "source": [
    "! gcsfuse --dir-mode 777 --file-mode 777 --implicit-dirs gp2_genotools_data ~/gp2_genotools_data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "study = 'VETPD'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "382\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>GP2sampleID_old</th>\n",
       "      <th>GP2sampleID</th>\n",
       "      <th>SentrixBarcode_A</th>\n",
       "      <th>SentrixPosition_A</th>\n",
       "      <th>study_old</th>\n",
       "      <th>study</th>\n",
       "      <th>sample_id</th>\n",
       "      <th>clinical_id</th>\n",
       "      <th>study_arm</th>\n",
       "      <th>study_type</th>\n",
       "      <th>...</th>\n",
       "      <th>family_history_other</th>\n",
       "      <th>family_history_pd</th>\n",
       "      <th>family_index</th>\n",
       "      <th>family_index_relationship</th>\n",
       "      <th>filename</th>\n",
       "      <th>FID</th>\n",
       "      <th>IID</th>\n",
       "      <th>pheno</th>\n",
       "      <th>sex_for_qc</th>\n",
       "      <th>GDPR</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>91402</th>\n",
       "      <td>NaN</td>\n",
       "      <td>VETPD_000001_s1</td>\n",
       "      <td>209042130073</td>\n",
       "      <td>R02C01</td>\n",
       "      <td>NaN</td>\n",
       "      <td>VETPD</td>\n",
       "      <td>GP2Z0001.1</td>\n",
       "      <td>GP2Z0001</td>\n",
       "      <td>Case</td>\n",
       "      <td>Case(/Control)</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>209042130073_R02C01</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>91403</th>\n",
       "      <td>NaN</td>\n",
       "      <td>VETPD_000002_s1</td>\n",
       "      <td>209042130008</td>\n",
       "      <td>R06C01</td>\n",
       "      <td>NaN</td>\n",
       "      <td>VETPD</td>\n",
       "      <td>GP2Z0002.1</td>\n",
       "      <td>GP2Z0002</td>\n",
       "      <td>Case</td>\n",
       "      <td>Case(/Control)</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>209042130008_R06C01</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>91404</th>\n",
       "      <td>NaN</td>\n",
       "      <td>VETPD_000003_s1</td>\n",
       "      <td>209042130073</td>\n",
       "      <td>R07C01</td>\n",
       "      <td>NaN</td>\n",
       "      <td>VETPD</td>\n",
       "      <td>GP2Z0003.1</td>\n",
       "      <td>GP2Z0003</td>\n",
       "      <td>Case</td>\n",
       "      <td>Case(/Control)</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>209042130073_R07C01</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>91405</th>\n",
       "      <td>NaN</td>\n",
       "      <td>VETPD_000004_s1</td>\n",
       "      <td>209042130062</td>\n",
       "      <td>R06C01</td>\n",
       "      <td>NaN</td>\n",
       "      <td>VETPD</td>\n",
       "      <td>GP2Z0004.1</td>\n",
       "      <td>GP2Z0004</td>\n",
       "      <td>Case</td>\n",
       "      <td>Case(/Control)</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>209042130062_R06C01</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>91406</th>\n",
       "      <td>NaN</td>\n",
       "      <td>VETPD_000005_s1</td>\n",
       "      <td>209042130055</td>\n",
       "      <td>R04C01</td>\n",
       "      <td>NaN</td>\n",
       "      <td>VETPD</td>\n",
       "      <td>GP2Z0005.2</td>\n",
       "      <td>GP2Z0005</td>\n",
       "      <td>Case</td>\n",
       "      <td>Case(/Control)</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>209042130055_R04C01</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows Ã— 51 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      GP2sampleID_old      GP2sampleID SentrixBarcode_A SentrixPosition_A  \\\n",
       "91402             NaN  VETPD_000001_s1     209042130073            R02C01   \n",
       "91403             NaN  VETPD_000002_s1     209042130008            R06C01   \n",
       "91404             NaN  VETPD_000003_s1     209042130073            R07C01   \n",
       "91405             NaN  VETPD_000004_s1     209042130062            R06C01   \n",
       "91406             NaN  VETPD_000005_s1     209042130055            R04C01   \n",
       "\n",
       "      study_old  study   sample_id clinical_id study_arm      study_type  ...  \\\n",
       "91402       NaN  VETPD  GP2Z0001.1    GP2Z0001      Case  Case(/Control)  ...   \n",
       "91403       NaN  VETPD  GP2Z0002.1    GP2Z0002      Case  Case(/Control)  ...   \n",
       "91404       NaN  VETPD  GP2Z0003.1    GP2Z0003      Case  Case(/Control)  ...   \n",
       "91405       NaN  VETPD  GP2Z0004.1    GP2Z0004      Case  Case(/Control)  ...   \n",
       "91406       NaN  VETPD  GP2Z0005.2    GP2Z0005      Case  Case(/Control)  ...   \n",
       "\n",
       "      family_history_other family_history_pd family_index  \\\n",
       "91402                  NaN               NaN          NaN   \n",
       "91403                  NaN               NaN          NaN   \n",
       "91404                  NaN               NaN          NaN   \n",
       "91405                  NaN               NaN          NaN   \n",
       "91406                  NaN               NaN          NaN   \n",
       "\n",
       "      family_index_relationship filename FID                  IID pheno  \\\n",
       "91402                       NaN      NaN   0  209042130073_R02C01     2   \n",
       "91403                       NaN      NaN   0  209042130008_R06C01     2   \n",
       "91404                       NaN      NaN   0  209042130073_R07C01     2   \n",
       "91405                       NaN      NaN   0  209042130062_R06C01     2   \n",
       "91406                       NaN      NaN   0  209042130055_R04C01     2   \n",
       "\n",
       "      sex_for_qc GDPR  \n",
       "91402          1    0  \n",
       "91403          1    0  \n",
       "91404          1    0  \n",
       "91405          1    0  \n",
       "91406          1    0  \n",
       "\n",
       "[5 rows x 51 columns]"
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "key = pd.read_csv(f'/home/levineks/gp2_genotools_data/clinical/GP2_master_key_MARCH_2025.txt', sep = '\\t', low_memory = False)\n",
    "key = key[key['study'] == study]\n",
    "print(len(key))\n",
    "key.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "VETPD\n",
      "(190, 6)\n",
      "   0                1  2  3  4  5\n",
      "0  0  VETPD_000028_s1  0  0  1 -9\n",
      "1  0  VETPD_000029_s1  0  0  1 -9\n",
      "2  0  VETPD_000030_s1  0  0  1 -9\n",
      "3  0  VETPD_000031_s1  0  0  1 -9\n",
      "4  0  VETPD_000032_s1  0  0  1 -9\n",
      "190\n"
     ]
    }
   ],
   "source": [
    "# If previous samples exist\n",
    "test = pd.read_csv(f'/home/levineks/gp2_genotools_data/merged_by_cohort_r10/GP2_merge_{study}.fam', sep = '\\s+', header = None)\n",
    "print(study)\n",
    "print(test.shape)\n",
    "print(test.head())\n",
    "list_included = list(test[1])\n",
    "print(len(list_included))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>GP2sampleID_old</th>\n",
       "      <th>GP2sampleID</th>\n",
       "      <th>SentrixBarcode_A</th>\n",
       "      <th>SentrixPosition_A</th>\n",
       "      <th>study_old</th>\n",
       "      <th>study</th>\n",
       "      <th>sample_id</th>\n",
       "      <th>clinical_id</th>\n",
       "      <th>study_arm</th>\n",
       "      <th>study_type</th>\n",
       "      <th>...</th>\n",
       "      <th>family_history_other</th>\n",
       "      <th>family_history_pd</th>\n",
       "      <th>family_index</th>\n",
       "      <th>family_index_relationship</th>\n",
       "      <th>filename</th>\n",
       "      <th>FID</th>\n",
       "      <th>IID</th>\n",
       "      <th>pheno</th>\n",
       "      <th>sex_for_qc</th>\n",
       "      <th>GDPR</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>91592</th>\n",
       "      <td>NaN</td>\n",
       "      <td>VETPD_000193_s1</td>\n",
       "      <td>209281700115</td>\n",
       "      <td>R01C01</td>\n",
       "      <td>NaN</td>\n",
       "      <td>VETPD</td>\n",
       "      <td>GP2Z0061.1</td>\n",
       "      <td>GP2Z0061</td>\n",
       "      <td>Case</td>\n",
       "      <td>Case(/Control)</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>209281700115_R01C01</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>91593</th>\n",
       "      <td>NaN</td>\n",
       "      <td>VETPD_000194_s1</td>\n",
       "      <td>209281700115</td>\n",
       "      <td>R02C01</td>\n",
       "      <td>NaN</td>\n",
       "      <td>VETPD</td>\n",
       "      <td>GP2Z0062.1</td>\n",
       "      <td>GP2Z0062</td>\n",
       "      <td>Case</td>\n",
       "      <td>Case(/Control)</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>209281700115_R02C01</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>91594</th>\n",
       "      <td>NaN</td>\n",
       "      <td>VETPD_000195_s1</td>\n",
       "      <td>209281700115</td>\n",
       "      <td>R03C01</td>\n",
       "      <td>NaN</td>\n",
       "      <td>VETPD</td>\n",
       "      <td>GP2Z0063.1</td>\n",
       "      <td>GP2Z0063</td>\n",
       "      <td>Case</td>\n",
       "      <td>Case(/Control)</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>209281700115_R03C01</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>91595</th>\n",
       "      <td>NaN</td>\n",
       "      <td>VETPD_000196_s1</td>\n",
       "      <td>209281700115</td>\n",
       "      <td>R04C01</td>\n",
       "      <td>NaN</td>\n",
       "      <td>VETPD</td>\n",
       "      <td>GP2Z0064.1</td>\n",
       "      <td>GP2Z0064</td>\n",
       "      <td>Case</td>\n",
       "      <td>Case(/Control)</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>209281700115_R04C01</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>91596</th>\n",
       "      <td>NaN</td>\n",
       "      <td>VETPD_000197_s1</td>\n",
       "      <td>209281700007</td>\n",
       "      <td>R06C01</td>\n",
       "      <td>NaN</td>\n",
       "      <td>VETPD</td>\n",
       "      <td>GP2Z0068.1</td>\n",
       "      <td>GP2Z0068</td>\n",
       "      <td>Case</td>\n",
       "      <td>Case(/Control)</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>209281700007_R06C01</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>91779</th>\n",
       "      <td>NaN</td>\n",
       "      <td>VETPD_000380_s1</td>\n",
       "      <td>209281710051</td>\n",
       "      <td>R08C01</td>\n",
       "      <td>NaN</td>\n",
       "      <td>VETPD</td>\n",
       "      <td>GP2Z0731.1</td>\n",
       "      <td>GP2Z0731</td>\n",
       "      <td>Case</td>\n",
       "      <td>Case(/Control)</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>209281710051_R08C01</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>91780</th>\n",
       "      <td>NaN</td>\n",
       "      <td>VETPD_000381_s1</td>\n",
       "      <td>209281710051</td>\n",
       "      <td>R07C01</td>\n",
       "      <td>NaN</td>\n",
       "      <td>VETPD</td>\n",
       "      <td>GP2Z0732.1</td>\n",
       "      <td>GP2Z0732</td>\n",
       "      <td>Case</td>\n",
       "      <td>Case(/Control)</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>209281710051_R07C01</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>91781</th>\n",
       "      <td>NaN</td>\n",
       "      <td>VETPD_000382_s1</td>\n",
       "      <td>209281710051</td>\n",
       "      <td>R06C01</td>\n",
       "      <td>NaN</td>\n",
       "      <td>VETPD</td>\n",
       "      <td>GP2Z0733.1</td>\n",
       "      <td>GP2Z0733</td>\n",
       "      <td>Case</td>\n",
       "      <td>Case(/Control)</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>209281710051_R06C01</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>91782</th>\n",
       "      <td>NaN</td>\n",
       "      <td>VETPD_000383_s1</td>\n",
       "      <td>209281710051</td>\n",
       "      <td>R05C01</td>\n",
       "      <td>NaN</td>\n",
       "      <td>VETPD</td>\n",
       "      <td>GP2Z0734.1</td>\n",
       "      <td>GP2Z0734</td>\n",
       "      <td>Case</td>\n",
       "      <td>Case(/Control)</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>209281710051_R05C01</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>91783</th>\n",
       "      <td>NaN</td>\n",
       "      <td>VETPD_000384_s1</td>\n",
       "      <td>209281700106</td>\n",
       "      <td>R02C01</td>\n",
       "      <td>NaN</td>\n",
       "      <td>VETPD</td>\n",
       "      <td>GP2Z0735.1</td>\n",
       "      <td>GP2Z0735</td>\n",
       "      <td>Case</td>\n",
       "      <td>Case(/Control)</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>209281700106_R02C01</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>192 rows Ã— 51 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      GP2sampleID_old      GP2sampleID SentrixBarcode_A SentrixPosition_A  \\\n",
       "91592             NaN  VETPD_000193_s1     209281700115            R01C01   \n",
       "91593             NaN  VETPD_000194_s1     209281700115            R02C01   \n",
       "91594             NaN  VETPD_000195_s1     209281700115            R03C01   \n",
       "91595             NaN  VETPD_000196_s1     209281700115            R04C01   \n",
       "91596             NaN  VETPD_000197_s1     209281700007            R06C01   \n",
       "...               ...              ...              ...               ...   \n",
       "91779             NaN  VETPD_000380_s1     209281710051            R08C01   \n",
       "91780             NaN  VETPD_000381_s1     209281710051            R07C01   \n",
       "91781             NaN  VETPD_000382_s1     209281710051            R06C01   \n",
       "91782             NaN  VETPD_000383_s1     209281710051            R05C01   \n",
       "91783             NaN  VETPD_000384_s1     209281700106            R02C01   \n",
       "\n",
       "      study_old  study   sample_id clinical_id study_arm      study_type  ...  \\\n",
       "91592       NaN  VETPD  GP2Z0061.1    GP2Z0061      Case  Case(/Control)  ...   \n",
       "91593       NaN  VETPD  GP2Z0062.1    GP2Z0062      Case  Case(/Control)  ...   \n",
       "91594       NaN  VETPD  GP2Z0063.1    GP2Z0063      Case  Case(/Control)  ...   \n",
       "91595       NaN  VETPD  GP2Z0064.1    GP2Z0064      Case  Case(/Control)  ...   \n",
       "91596       NaN  VETPD  GP2Z0068.1    GP2Z0068      Case  Case(/Control)  ...   \n",
       "...         ...    ...         ...         ...       ...             ...  ...   \n",
       "91779       NaN  VETPD  GP2Z0731.1    GP2Z0731      Case  Case(/Control)  ...   \n",
       "91780       NaN  VETPD  GP2Z0732.1    GP2Z0732      Case  Case(/Control)  ...   \n",
       "91781       NaN  VETPD  GP2Z0733.1    GP2Z0733      Case  Case(/Control)  ...   \n",
       "91782       NaN  VETPD  GP2Z0734.1    GP2Z0734      Case  Case(/Control)  ...   \n",
       "91783       NaN  VETPD  GP2Z0735.1    GP2Z0735      Case  Case(/Control)  ...   \n",
       "\n",
       "      family_history_other family_history_pd family_index  \\\n",
       "91592                  NaN               NaN          NaN   \n",
       "91593                  NaN               NaN          NaN   \n",
       "91594                  NaN               NaN          NaN   \n",
       "91595                  NaN               NaN          NaN   \n",
       "91596                  NaN               NaN          NaN   \n",
       "...                    ...               ...          ...   \n",
       "91779                  NaN               NaN          NaN   \n",
       "91780                  NaN               NaN          NaN   \n",
       "91781                  NaN               NaN          NaN   \n",
       "91782                  NaN               NaN          NaN   \n",
       "91783                  NaN               NaN          NaN   \n",
       "\n",
       "      family_index_relationship filename FID                  IID pheno  \\\n",
       "91592                       NaN      NaN   0  209281700115_R01C01     2   \n",
       "91593                       NaN      NaN   0  209281700115_R02C01     2   \n",
       "91594                       NaN      NaN   0  209281700115_R03C01     2   \n",
       "91595                       NaN      NaN   0  209281700115_R04C01     2   \n",
       "91596                       NaN      NaN   0  209281700007_R06C01     2   \n",
       "...                         ...      ...  ..                  ...   ...   \n",
       "91779                       NaN      NaN   0  209281710051_R08C01     2   \n",
       "91780                       NaN      NaN   0  209281710051_R07C01     2   \n",
       "91781                       NaN      NaN   0  209281710051_R06C01     2   \n",
       "91782                       NaN      NaN   0  209281710051_R05C01     2   \n",
       "91783                       NaN      NaN   0  209281700106_R02C01     2   \n",
       "\n",
       "      sex_for_qc GDPR  \n",
       "91592          1    0  \n",
       "91593          1    0  \n",
       "91594          1    0  \n",
       "91595          1    0  \n",
       "91596          1    0  \n",
       "...          ...  ...  \n",
       "91779          1    0  \n",
       "91780          1    0  \n",
       "91781          1    0  \n",
       "91782          1    0  \n",
       "91783          1    0  \n",
       "\n",
       "[192 rows x 51 columns]"
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#The samples we still need to process are the ones that were NOT included in the prevous merged file\n",
    "key = key[~key['GP2sampleID'].isin(list_included)]\n",
    "key"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "192\n",
      "24\n",
      "['209281700022', '209281700020', '209281710133', '209281700092', '209281700007', '209281700115', '209281700035', '209281700163', '209281700041', '209281710005', '209281700017', '209281710051', '209281700145', '209281700123', '209281700112', '209281700106', '209281710050', '209281700107', '209281700069', '209281710135', '209281700119', '209281700122', '209281700040', '209281700084']\n"
     ]
    }
   ],
   "source": [
    "print(len(key))\n",
    "barcode_list = list(set(list(key['SentrixBarcode_A'])))\n",
    "print(len(barcode_list))\n",
    "print(barcode_list)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Preparing bash script to convert IDATs to ped for VETPD.\n"
     ]
    }
   ],
   "source": [
    "# Create shell script to execute -- convert IDATs to ped\n",
    "count = 0 # for Job ID count\n",
    "\n",
    "# Establish paths for inputs/outputs\n",
    "\n",
    "iaap = f'./gp2_genotools_data/ilmn_files/iaap-cli/iaap-cli'\n",
    "bpm = f'./gp2_genotools_data/ilmn_files/NeuroBooster_20042459_A2.bpm'\n",
    "egt = f'./gp2_genotools_data/ilmn_files/recluster_09272022.egt'\n",
    "raw_plink_path = f'./gp2_genotools_data/ped_bed'\n",
    "idat_path = f'./gp2_idats/${{BARCODE}}'\n",
    "\n",
    "idat_to_ped_cmd = f'\\\n",
    "{iaap} gencall \\\n",
    "{bpm} \\\n",
    "{egt} \\\n",
    "{raw_plink_path}/ \\\n",
    "-f {idat_path} \\\n",
    "-p \\\n",
    "-t 8'\n",
    "\n",
    "with open(f'./gp2_genotools_data/batch_files/convert_idats_to_ped.sh', 'w') as f:\n",
    "    f.write('#!/bin/bash\\n\\n')\n",
    "    f.write('BARCODE=$1\\n\\n')\n",
    "    f.write(f'{idat_to_ped_cmd}\\n')\n",
    "    f.close()\n",
    "\n",
    "print(f\"Preparing bash script to convert IDATs to ped for {study}.\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [],
   "source": [
    "# 2 buckets mounted: 1 for inputs, 1 for outputs\n",
    "def create_script_job_with_buckets(\n",
    "    project_id: str, region: str, job_name: str, bucket_name_input: str, bucket_name_output: str, script_text: str) -> batch_v1.Job:\n",
    "    \"\"\"\n",
    "    This method shows how to create a sample Batch Job that will run\n",
    "    a simple command on Cloud Compute instances.\n",
    "\n",
    "    Args:\n",
    "        project_id: project ID or project number of the Cloud project you want to use.\n",
    "        region: name of the region you want to use to run the job. Regions that are\n",
    "            available for Batch are listed on: https://cloud.google.com/batch/docs/get-started#locations\n",
    "        job_name: the name of the job that will be created.\n",
    "            It needs to be unique for each project and region pair.\n",
    "        bucket_name: name of the bucket to be mounted for your Job.\n",
    "\n",
    "    Returns:\n",
    "        A job object representing the job created.\n",
    "    \"\"\"\n",
    "    client = batch_v1.BatchServiceClient()\n",
    "\n",
    "    # Define what will be done as part of the job.\n",
    "    task = batch_v1.TaskSpec()\n",
    "    runnable = batch_v1.Runnable()\n",
    "\n",
    "    # This would replace runnable script\n",
    "    # runnable.container = batch_v1.Runnable.Container()\n",
    "    # runnable.container.image_uri = docker_image\n",
    "\n",
    "    runnable.script = batch_v1.Runnable.Script()\n",
    "    runnable.script.text = script_text\n",
    "    task.runnables = [runnable]\n",
    "\n",
    "    # Define the first GCS bucket \n",
    "    gcs_bucket_input = batch_v1.GCS(remote_path=bucket_name_input)\n",
    "    gcs_volume_input = batch_v1.Volume(gcs=gcs_bucket_input, mount_path=\"./gp2_idats\")\n",
    "\n",
    "    # Define the second GCS bucket \n",
    "    gcs_bucket_output = batch_v1.GCS(remote_path=bucket_name_output)\n",
    "    gcs_volume_output = batch_v1.Volume(gcs=gcs_bucket_output, mount_path=\"./gp2_genotools_data\")\n",
    "\n",
    "    # Add both volumes to the task\n",
    "    task.volumes = [gcs_volume_input, gcs_volume_output]\n",
    "\n",
    "    # We can specify what resources are requested by each task.\n",
    "    resources = batch_v1.ComputeResource()\n",
    "    resources.cpu_milli = 8000  #in milliseconds per cpu-second. This means the task requires 50% of a single CPUs; 2000 = 2 CPUs\n",
    "    resources.memory_mib = 32768 # 4 GiB of memory (4096 MiB - mebibyte); 32768 MiB = 32 GB; 16384 = 16 GB\n",
    "    task.compute_resource = resources\n",
    "\n",
    "    task.max_retry_count = 0 # I'd suggest changing this to 0; otherwise, it tries two times before it fails\n",
    "    task.max_run_duration = \"3600s\" # 24 hours = 86400s; 1 hour = 3600s; 8 hours = 28800s\n",
    "\n",
    "    # Tasks are grouped inside a job using TaskGroups.\n",
    "    # Currently, it's possible to have only one task group.\n",
    "    group = batch_v1.TaskGroup()\n",
    "    group.task_count = 1\n",
    "    group.task_spec = task\n",
    "\n",
    "    # Policies are used to define on what kind of virtual machines the tasks will run on.\n",
    "    # Read more about machine types here: https://cloud.google.com/compute/docs/machine-types\n",
    "    allocation_policy = batch_v1.AllocationPolicy()\n",
    "    policy = batch_v1.AllocationPolicy.InstancePolicy()\n",
    "    policy.machine_type = \"e2-standard-8\"\n",
    "    instances = batch_v1.AllocationPolicy.InstancePolicyOrTemplate()\n",
    "    instances.policy = policy\n",
    "    allocation_policy.instances = [instances]\n",
    "\n",
    "    job = batch_v1.Job()\n",
    "    job.task_groups = [group]\n",
    "    job.allocation_policy = allocation_policy\n",
    "    job.labels = {\"env\": \"testing\", \"type\": \"script\", \"mount\": \"bucket\"}\n",
    "    # We use Cloud Logging as it's an out of the box available option\n",
    "    job.logs_policy = batch_v1.LogsPolicy()\n",
    "    job.logs_policy.destination = batch_v1.LogsPolicy.Destination.CLOUD_LOGGING\n",
    "\n",
    "    create_request = batch_v1.CreateJobRequest()\n",
    "    create_request.job = job\n",
    "    create_request.job_id = job_name\n",
    "    # The job's parent is the region in which the job will run\n",
    "    create_request.parent = f\"projects/{project_id}/locations/{region}\"\n",
    "\n",
    "    return client.create_job(create_request)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "#!/bin/bash\n",
      "\n",
      "BARCODE=$1\n",
      "\n",
      "./gp2_genotools_data/ilmn_files/iaap-cli/iaap-cli gencall ./gp2_genotools_data/ilmn_files/NeuroBooster_20042459_A2.bpm ./gp2_genotools_data/ilmn_files/recluster_09272022.egt ./gp2_genotools_data/ped_bed/ -f ./gp2_idats/${BARCODE} -p -t 8\n"
     ]
    }
   ],
   "source": [
    "# Double check the shell script to make sure it does what you need and paths are correct\n",
    "! cat ./gp2_genotools_data/batch_files/convert_idats_to_ped.sh"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "24\n",
      "['209281700022', '209281700020', '209281710133', '209281700092', '209281700007', '209281700115', '209281700035', '209281700163', '209281700041', '209281710005', '209281700017', '209281710051', '209281700145', '209281700123', '209281700112', '209281700106', '209281710050', '209281700107', '209281700069', '209281710135', '209281700119', '209281700122', '209281700040', '209281700084']\n"
     ]
    }
   ],
   "source": [
    "barcode_list = list(set(list(key['SentrixBarcode_A'])))\n",
    "print(len(barcode_list))\n",
    "print(barcode_list)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'VETPD'"
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "study"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Update the count with the last job # that you ran:\n",
    "count = 10\n",
    "\n",
    "# Define the number of codes to process per job (8)\n",
    "codes_per_job = 5\n",
    "\n",
    "# Split the filename_list into chunks\n",
    "from itertools import zip_longest\n",
    "\n",
    "# Helper function to chunk the list into groups of 8\n",
    "def chunk_list(iterable, n):\n",
    "    args = [iter(iterable)] * n\n",
    "    return zip_longest(*args)\n",
    "\n",
    "# Loop through each chunk of 8 codes and create a single job\n",
    "for chunk in chunk_list(barcode_list, codes_per_job):\n",
    "    # Filter out any None values from the last incomplete chunk\n",
    "    codes = [code for code in chunk if code is not None]\n",
    "    \n",
    "    # Generate a unique job name\n",
    "    count += 1\n",
    "    job_name = f'idattoped{study.lower()}{count}'\n",
    "    \n",
    "    # Create a combined script for the N codes\n",
    "    script = f\"\"\"\n",
    "        #!/bin/bash\n",
    "\n",
    "        # Install and activate PLINK\n",
    "        chmod +x ./gp2_genotools_data/batch_files/plink2_module.sh\n",
    "        ./gp2_genotools_data/batch_files/plink2_module.sh\n",
    "    \"\"\"\n",
    "\n",
    "    # Add commands for each code in the chunk\n",
    "    for code in codes:\n",
    "        script += f\"\"\"\n",
    "        # Make analysis script executable and run for a specific code\n",
    "        chmod +x ./gp2_genotools_data/batch_files/convert_idats_to_ped.sh\n",
    "        ./gp2_genotools_data/batch_files/convert_idats_to_ped.sh {code}\n",
    "        \"\"\"\n",
    "\n",
    "    # Create the job with the combined script\n",
    "    create_script_job_with_buckets(\n",
    "        project_id = \"gp2-release-terra\", \n",
    "        region = 'europe-west4', \n",
    "        job_name = job_name, \n",
    "        bucket_name_input= 'gp2_idats', bucket_name_output= 'gp2_genotools_data', \n",
    "        script_text = script\n",
    "    )"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "VETPD\n",
      "Last job: 15\n",
      "Job Status: SUCCEEDED\n"
     ]
    }
   ],
   "source": [
    "# Initialize the client\n",
    "#count = 1\n",
    "client = batch_v1.BatchServiceClient()\n",
    "job_id = f'idattoped{study.lower()}{count}'\n",
    "\n",
    "# Define the job name (replace with your project ID, location, and job ID)\n",
    "project_id = \"gp2-release-terra\"\n",
    "region = 'europe-west4'\n",
    "job_name = f\"projects/{project_id}/locations/{region}/jobs/{job_id}\"\n",
    "\n",
    "# Get the job details\n",
    "job = client.get_job(name=job_name)\n",
    "\n",
    "# Check the status\n",
    "job_status = job.status.state\n",
    "print(study)\n",
    "print(\"Last job:\", count)\n",
    "print(f\"Job Status: {batch_v1.JobStatus.State(job_status).name}\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [],
   "source": [
    "# can also check on job here: https://console.cloud.google.com/batch/jobs?project=gp2-release-terra&inv=1&invt=Abj2Qw"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## check if any ped job has failed"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "VETPD\n",
      "192\n"
     ]
    }
   ],
   "source": [
    "study_list = [study]\n",
    "print(study)\n",
    "print(len(key))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "There is 0 missing ped file for VETPD.\n"
     ]
    }
   ],
   "source": [
    "# copy map file to match name of each ped\n",
    "missing_peds = []\n",
    "map_file = f'./gp2_genotools_data/ped_bed/NeuroBooster_20042459_A2.map'\n",
    "raw_plink_path = f'./gp2_genotools_data/ped_bed'\n",
    "missing_idat_dir = f'./gp2_genotools_data/missing'\n",
    "\n",
    "for cohort in study_list:\n",
    "    df = key[key['study']==cohort]\n",
    "    \n",
    "    missing_cnt = 0\n",
    "\n",
    "    for filename in df.IID:\n",
    "        ped = f'{raw_plink_path}/{filename}.ped'\n",
    "        out_map = f'{raw_plink_path}/{filename}.map'\n",
    "        if os.path.isfile(ped):\n",
    "            shutil.copyfile(src=map_file, dst=out_map)\n",
    "        else:\n",
    "            missing_cnt += 1\n",
    "            missing_peds.append(filename)\n",
    "            \n",
    "    print(f\"There is {missing_cnt} missing ped file for {cohort}.\")\n",
    "\n",
    "with open(f'{missing_idat_dir}/missing_peds_{cohort}.txt', 'w') as f:\n",
    "    for m_ped in missing_peds:\n",
    "        f.write(f'{m_ped}\\n')\n",
    "f.close()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Series([], Name: SentrixBarcode_A, dtype: int64)"
      ]
     },
     "execution_count": 24,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "missing = key[key['IID'].isin(missing_peds)]\n",
    "missing.SentrixBarcode_A.value_counts()\n",
    "#missing.to_csv('/home/levineks/gp2_genotools_data/missing/SP-BCN2_n8_missing.csv', header = True, index = False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>GP2sampleID_old</th>\n",
       "      <th>GP2sampleID</th>\n",
       "      <th>SentrixBarcode_A</th>\n",
       "      <th>SentrixPosition_A</th>\n",
       "      <th>study_old</th>\n",
       "      <th>study</th>\n",
       "      <th>sample_id</th>\n",
       "      <th>clinical_id</th>\n",
       "      <th>study_arm</th>\n",
       "      <th>study_type</th>\n",
       "      <th>...</th>\n",
       "      <th>family_history_other</th>\n",
       "      <th>family_history_pd</th>\n",
       "      <th>family_index</th>\n",
       "      <th>family_index_relationship</th>\n",
       "      <th>filename</th>\n",
       "      <th>FID</th>\n",
       "      <th>IID</th>\n",
       "      <th>pheno</th>\n",
       "      <th>sex_for_qc</th>\n",
       "      <th>GDPR</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>91592</th>\n",
       "      <td>NaN</td>\n",
       "      <td>VETPD_000193_s1</td>\n",
       "      <td>209281700115</td>\n",
       "      <td>R01C01</td>\n",
       "      <td>NaN</td>\n",
       "      <td>VETPD</td>\n",
       "      <td>GP2Z0061.1</td>\n",
       "      <td>GP2Z0061</td>\n",
       "      <td>Case</td>\n",
       "      <td>Case(/Control)</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>209281700115_R01C01</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>91593</th>\n",
       "      <td>NaN</td>\n",
       "      <td>VETPD_000194_s1</td>\n",
       "      <td>209281700115</td>\n",
       "      <td>R02C01</td>\n",
       "      <td>NaN</td>\n",
       "      <td>VETPD</td>\n",
       "      <td>GP2Z0062.1</td>\n",
       "      <td>GP2Z0062</td>\n",
       "      <td>Case</td>\n",
       "      <td>Case(/Control)</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>209281700115_R02C01</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>91594</th>\n",
       "      <td>NaN</td>\n",
       "      <td>VETPD_000195_s1</td>\n",
       "      <td>209281700115</td>\n",
       "      <td>R03C01</td>\n",
       "      <td>NaN</td>\n",
       "      <td>VETPD</td>\n",
       "      <td>GP2Z0063.1</td>\n",
       "      <td>GP2Z0063</td>\n",
       "      <td>Case</td>\n",
       "      <td>Case(/Control)</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>209281700115_R03C01</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>91595</th>\n",
       "      <td>NaN</td>\n",
       "      <td>VETPD_000196_s1</td>\n",
       "      <td>209281700115</td>\n",
       "      <td>R04C01</td>\n",
       "      <td>NaN</td>\n",
       "      <td>VETPD</td>\n",
       "      <td>GP2Z0064.1</td>\n",
       "      <td>GP2Z0064</td>\n",
       "      <td>Case</td>\n",
       "      <td>Case(/Control)</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>209281700115_R04C01</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>91596</th>\n",
       "      <td>NaN</td>\n",
       "      <td>VETPD_000197_s1</td>\n",
       "      <td>209281700007</td>\n",
       "      <td>R06C01</td>\n",
       "      <td>NaN</td>\n",
       "      <td>VETPD</td>\n",
       "      <td>GP2Z0068.1</td>\n",
       "      <td>GP2Z0068</td>\n",
       "      <td>Case</td>\n",
       "      <td>Case(/Control)</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>209281700007_R06C01</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>91779</th>\n",
       "      <td>NaN</td>\n",
       "      <td>VETPD_000380_s1</td>\n",
       "      <td>209281710051</td>\n",
       "      <td>R08C01</td>\n",
       "      <td>NaN</td>\n",
       "      <td>VETPD</td>\n",
       "      <td>GP2Z0731.1</td>\n",
       "      <td>GP2Z0731</td>\n",
       "      <td>Case</td>\n",
       "      <td>Case(/Control)</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>209281710051_R08C01</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>91780</th>\n",
       "      <td>NaN</td>\n",
       "      <td>VETPD_000381_s1</td>\n",
       "      <td>209281710051</td>\n",
       "      <td>R07C01</td>\n",
       "      <td>NaN</td>\n",
       "      <td>VETPD</td>\n",
       "      <td>GP2Z0732.1</td>\n",
       "      <td>GP2Z0732</td>\n",
       "      <td>Case</td>\n",
       "      <td>Case(/Control)</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>209281710051_R07C01</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>91781</th>\n",
       "      <td>NaN</td>\n",
       "      <td>VETPD_000382_s1</td>\n",
       "      <td>209281710051</td>\n",
       "      <td>R06C01</td>\n",
       "      <td>NaN</td>\n",
       "      <td>VETPD</td>\n",
       "      <td>GP2Z0733.1</td>\n",
       "      <td>GP2Z0733</td>\n",
       "      <td>Case</td>\n",
       "      <td>Case(/Control)</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>209281710051_R06C01</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>91782</th>\n",
       "      <td>NaN</td>\n",
       "      <td>VETPD_000383_s1</td>\n",
       "      <td>209281710051</td>\n",
       "      <td>R05C01</td>\n",
       "      <td>NaN</td>\n",
       "      <td>VETPD</td>\n",
       "      <td>GP2Z0734.1</td>\n",
       "      <td>GP2Z0734</td>\n",
       "      <td>Case</td>\n",
       "      <td>Case(/Control)</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>209281710051_R05C01</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>91783</th>\n",
       "      <td>NaN</td>\n",
       "      <td>VETPD_000384_s1</td>\n",
       "      <td>209281700106</td>\n",
       "      <td>R02C01</td>\n",
       "      <td>NaN</td>\n",
       "      <td>VETPD</td>\n",
       "      <td>GP2Z0735.1</td>\n",
       "      <td>GP2Z0735</td>\n",
       "      <td>Case</td>\n",
       "      <td>Case(/Control)</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>209281700106_R02C01</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>192 rows Ã— 51 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      GP2sampleID_old      GP2sampleID SentrixBarcode_A SentrixPosition_A  \\\n",
       "91592             NaN  VETPD_000193_s1     209281700115            R01C01   \n",
       "91593             NaN  VETPD_000194_s1     209281700115            R02C01   \n",
       "91594             NaN  VETPD_000195_s1     209281700115            R03C01   \n",
       "91595             NaN  VETPD_000196_s1     209281700115            R04C01   \n",
       "91596             NaN  VETPD_000197_s1     209281700007            R06C01   \n",
       "...               ...              ...              ...               ...   \n",
       "91779             NaN  VETPD_000380_s1     209281710051            R08C01   \n",
       "91780             NaN  VETPD_000381_s1     209281710051            R07C01   \n",
       "91781             NaN  VETPD_000382_s1     209281710051            R06C01   \n",
       "91782             NaN  VETPD_000383_s1     209281710051            R05C01   \n",
       "91783             NaN  VETPD_000384_s1     209281700106            R02C01   \n",
       "\n",
       "      study_old  study   sample_id clinical_id study_arm      study_type  ...  \\\n",
       "91592       NaN  VETPD  GP2Z0061.1    GP2Z0061      Case  Case(/Control)  ...   \n",
       "91593       NaN  VETPD  GP2Z0062.1    GP2Z0062      Case  Case(/Control)  ...   \n",
       "91594       NaN  VETPD  GP2Z0063.1    GP2Z0063      Case  Case(/Control)  ...   \n",
       "91595       NaN  VETPD  GP2Z0064.1    GP2Z0064      Case  Case(/Control)  ...   \n",
       "91596       NaN  VETPD  GP2Z0068.1    GP2Z0068      Case  Case(/Control)  ...   \n",
       "...         ...    ...         ...         ...       ...             ...  ...   \n",
       "91779       NaN  VETPD  GP2Z0731.1    GP2Z0731      Case  Case(/Control)  ...   \n",
       "91780       NaN  VETPD  GP2Z0732.1    GP2Z0732      Case  Case(/Control)  ...   \n",
       "91781       NaN  VETPD  GP2Z0733.1    GP2Z0733      Case  Case(/Control)  ...   \n",
       "91782       NaN  VETPD  GP2Z0734.1    GP2Z0734      Case  Case(/Control)  ...   \n",
       "91783       NaN  VETPD  GP2Z0735.1    GP2Z0735      Case  Case(/Control)  ...   \n",
       "\n",
       "      family_history_other family_history_pd family_index  \\\n",
       "91592                  NaN               NaN          NaN   \n",
       "91593                  NaN               NaN          NaN   \n",
       "91594                  NaN               NaN          NaN   \n",
       "91595                  NaN               NaN          NaN   \n",
       "91596                  NaN               NaN          NaN   \n",
       "...                    ...               ...          ...   \n",
       "91779                  NaN               NaN          NaN   \n",
       "91780                  NaN               NaN          NaN   \n",
       "91781                  NaN               NaN          NaN   \n",
       "91782                  NaN               NaN          NaN   \n",
       "91783                  NaN               NaN          NaN   \n",
       "\n",
       "      family_index_relationship filename FID                  IID pheno  \\\n",
       "91592                       NaN      NaN   0  209281700115_R01C01     2   \n",
       "91593                       NaN      NaN   0  209281700115_R02C01     2   \n",
       "91594                       NaN      NaN   0  209281700115_R03C01     2   \n",
       "91595                       NaN      NaN   0  209281700115_R04C01     2   \n",
       "91596                       NaN      NaN   0  209281700007_R06C01     2   \n",
       "...                         ...      ...  ..                  ...   ...   \n",
       "91779                       NaN      NaN   0  209281710051_R08C01     2   \n",
       "91780                       NaN      NaN   0  209281710051_R07C01     2   \n",
       "91781                       NaN      NaN   0  209281710051_R06C01     2   \n",
       "91782                       NaN      NaN   0  209281710051_R05C01     2   \n",
       "91783                       NaN      NaN   0  209281700106_R02C01     2   \n",
       "\n",
       "      sex_for_qc GDPR  \n",
       "91592          1    0  \n",
       "91593          1    0  \n",
       "91594          1    0  \n",
       "91595          1    0  \n",
       "91596          1    0  \n",
       "...          ...  ...  \n",
       "91779          1    0  \n",
       "91780          1    0  \n",
       "91781          1    0  \n",
       "91782          1    0  \n",
       "91783          1    0  \n",
       "\n",
       "[192 rows x 51 columns]"
      ]
     },
     "execution_count": 25,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "key = key[~key['IID'].isin(missing_peds)]\n",
    "key"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "VETPD    192\n",
       "Name: study, dtype: int64"
      ]
     },
     "execution_count": 26,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "key.study.value_counts()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Create bed files"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {},
   "outputs": [],
   "source": [
    "# 1 bucket mounted for inputs/ouput\n",
    "\n",
    "def create_script_job_with_bucket(\n",
    "    project_id: str, region: str, job_name: str, bucket_name: str, script_text: str) -> batch_v1.Job:\n",
    "    \"\"\"\n",
    "    This method shows how to create a sample Batch Job that will run\n",
    "    a simple command on Cloud Compute instances.\n",
    "\n",
    "    Args:\n",
    "        project_id: project ID or project number of the Cloud project you want to use.\n",
    "        region: name of the region you want to use to run the job. Regions that are\n",
    "            available for Batch are listed on: https://cloud.google.com/batch/docs/get-started#locations\n",
    "        job_name: the name of the job that will be created.\n",
    "            It needs to be unique for each project and region pair.\n",
    "        bucket_name: name of the bucket to be mounted for your Job.\n",
    "\n",
    "    Returns:\n",
    "        A job object representing the job created.\n",
    "    \"\"\"\n",
    "    client = batch_v1.BatchServiceClient()\n",
    "\n",
    "    # Define what will be done as part of the job.\n",
    "    task = batch_v1.TaskSpec()\n",
    "    runnable = batch_v1.Runnable()\n",
    "    runnable.script = batch_v1.Runnable.Script()\n",
    "    runnable.script.text = script_text\n",
    "    task.runnables = [runnable]\n",
    "\n",
    "    gcs_bucket = batch_v1.GCS()\n",
    "    gcs_bucket.remote_path = bucket_name\n",
    "    gcs_volume = batch_v1.Volume()\n",
    "    gcs_volume.gcs = gcs_bucket\n",
    "    gcs_volume.mount_path = \"./gp2_genotools_data\"\n",
    "    task.volumes = [gcs_volume]\n",
    "\n",
    "    # We can specify what resources are requested by each task.\n",
    "    resources = batch_v1.ComputeResource()\n",
    "    resources.cpu_milli = 8000  #8000 milliseconds = 8 CPUs\n",
    "    resources.memory_mib = 32768 # 200192 MiB = 200 GB of memory; 32768 MiB = 32 GB; 16384 = 16 GB\n",
    "    task.compute_resource = resources\n",
    "\n",
    "    task.max_retry_count = 0\n",
    "    #task.max_run_duration = \"3600s\" # 24 hours = 86400s; 1 hour = 3600s; 8 hours = 28800s\n",
    "    task.max_run_duration = \"28800s\" # 24 hours = 86400s; 1 hour = 3600s; 8 hours = 28800s\n",
    "\n",
    "    # Tasks are grouped inside a job using TaskGroups.\n",
    "    # Currently, it's possible to have only one task group.\n",
    "    group = batch_v1.TaskGroup()\n",
    "    group.task_count = 1\n",
    "    group.task_spec = task\n",
    "\n",
    "    # Policies are used to define on what kind of virtual machines the tasks will run on.\n",
    "    # In this case, we tell the system to use \"e2-standard-4\" machine type.\n",
    "    # Read more about machine types here: https://cloud.google.com/compute/docs/machine-types\n",
    "    allocation_policy = batch_v1.AllocationPolicy()\n",
    "    policy = batch_v1.AllocationPolicy.InstancePolicy()\n",
    "    policy.machine_type = \"e2-standard-8\"\n",
    "    instances = batch_v1.AllocationPolicy.InstancePolicyOrTemplate()\n",
    "    instances.policy = policy\n",
    "    allocation_policy.instances = [instances]\n",
    "\n",
    "    job = batch_v1.Job()\n",
    "    job.task_groups = [group]\n",
    "    job.allocation_policy = allocation_policy\n",
    "    job.labels = {\"env\": \"testing\", \"type\": \"script\", \"mount\": \"bucket\"}\n",
    "    # We use Cloud Logging as it's an out of the box available option\n",
    "    job.logs_policy = batch_v1.LogsPolicy()\n",
    "    job.logs_policy.destination = batch_v1.LogsPolicy.Destination.CLOUD_LOGGING\n",
    "\n",
    "    create_request = batch_v1.CreateJobRequest()\n",
    "    create_request.job = job\n",
    "    create_request.job_id = job_name\n",
    "    # The job's parent is the region in which the job will run\n",
    "    create_request.parent = f\"projects/{project_id}/locations/{region}\"\n",
    "\n",
    "    return client.create_job(create_request)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Preparing bash script to convert ped to bed for VETPD.\n"
     ]
    }
   ],
   "source": [
    "# Create shell script to execute -- convert ped to bed\n",
    "count = 0 # for Job ID count\n",
    "\n",
    "# Establish paths for inputs/outputs\n",
    "\n",
    "ped_file = f'./gp2_genotools_data/ped_bed/${{FILENAME}}'\n",
    "\n",
    "make_bed_cmd = f\"/home/levineks/bin/plink1.9/plink --file {ped_file} --make-bed --out {ped_file}\"\n",
    "\n",
    "with open(f'./gp2_genotools_data/batch_files/convert_ped_to_bed.sh', 'w') as f:\n",
    "    f.write('#!/bin/bash\\n\\n')\n",
    "    f.write('FILENAME=$1\\n\\n')\n",
    "    f.write(f'{make_bed_cmd}\\n')\n",
    "    f.close()\n",
    "\n",
    "print(f\"Preparing bash script to convert ped to bed for {study}.\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "#!/bin/bash\n",
      "\n",
      "FILENAME=$1\n",
      "\n",
      "/home/levineks/bin/plink1.9/plink --file ./gp2_genotools_data/ped_bed/${FILENAME} --make-bed --out ./gp2_genotools_data/ped_bed/${FILENAME}\n"
     ]
    }
   ],
   "source": [
    "# Double check the shell script to make sure it does what you need and paths are correct\n",
    "! cat ./gp2_genotools_data/batch_files/convert_ped_to_bed.sh"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "192\n",
      "['209281700115_R01C01', '209281700115_R02C01', '209281700115_R03C01', '209281700115_R04C01', '209281700007_R06C01', '209281700115_R05C01', '209281700007_R07C01', '209281700007_R08C01', '209281700040_R01C01', '209281700115_R06C01', '209281700115_R07C01', '209281700115_R08C01', '209281700092_R01C01', '209281700092_R02C01', '209281700092_R03C01', '209281700041_R08C01', '209281710051_R01C01', '209281700092_R04C01', '209281700092_R05C01', '209281700040_R04C01', '209281700092_R06C01', '209281700040_R05C01', '209281700092_R07C01', '209281700122_R02C01', '209281700092_R08C01', '209281700122_R01C01', '209281700040_R06C01', '209281700040_R07C01', '209281700035_R04C01', '209281700035_R05C01', '209281700122_R03C01', '209281700007_R05C01', '209281700040_R08C01', '209281700122_R04C01', '209281700084_R01C01', '209281710133_R01C01', '209281710133_R02C01', '209281700084_R02C01', '209281700040_R02C01', '209281700122_R05C01', '209281700122_R06C01', '209281700084_R03C01', '209281710133_R03C01', '209281710133_R04C01', '209281700035_R06C01', '209281700122_R07C01', '209281700084_R05C01', '209281700084_R04C01', '209281700084_R06C01', '209281700035_R07C01', '209281700035_R08C01', '209281700017_R01C01', '209281700084_R07C01', '209281710133_R05C01', '209281710133_R07C01', '209281710133_R06C01', '209281700084_R08C01', '209281700122_R08C01', '209281700020_R01C01', '209281700020_R02C01', '209281700020_R03C01', '209281710133_R08C01', '209281700119_R01C01', '209281700119_R02C01', '209281700119_R03C01', '209281700017_R02C01', '209281700017_R03C01', '209281700119_R04C01', '209281700119_R05C01', '209281700145_R01C01', '209281700145_R02C01', '209281700145_R03C01', '209281700145_R04C01', '209281700119_R06C01', '209281700020_R04C01', '209281700106_R01C01', '209281700020_R05C01', '209281700119_R07C01', '209281700145_R05C01', '209281700119_R08C01', '209281700069_R01C01', '209281700020_R06C01', '209281700145_R06C01', '209281700145_R07C01', '209281700145_R08C01', '209281710050_R01C01', '209281710050_R02C01', '209281710050_R03C01', '209281710050_R04C01', '209281700020_R07C01', '209281700020_R08C01', '209281700022_R01C01', '209281710050_R05C01', '209281700017_R04C01', '209281700017_R05C01', '209281700017_R06C01', '209281700017_R07C01', '209281710050_R06C01', '209281710050_R07C01', '209281710050_R08C01', '209281700112_R01C01', '209281700022_R02C01', '209281700022_R03C01', '209281700022_R04C01', '209281700112_R02C01', '209281700112_R03C01', '209281700112_R04C01', '209281700069_R02C01', '209281710135_R02C01', '209281700069_R03C01', '209281710135_R03C01', '209281700112_R05C01', '209281700022_R05C01', '209281700112_R06C01', '209281700112_R07C01', '209281700069_R04C01', '209281700022_R06C01', '209281700112_R08C01', '209281700022_R07C01', '209281700163_R01C01', '209281700163_R02C01', '209281700163_R03C01', '209281700163_R04C01', '209281700069_R05C01', '209281700163_R05C01', '209281700069_R06C01', '209281700163_R06C01', '209281710135_R04C01', '209281710135_R05C01', '209281710135_R06C01', '209281700041_R04C01', '209281710135_R07C01', '209281700163_R07C01', '209281700069_R07C01', '209281700163_R08C01', '209281700069_R08C01', '209281700123_R01C01', '209281710135_R08C01', '209281700041_R03C01', '209281700017_R08C01', '209281700007_R01C01', '209281700007_R02C01', '209281700007_R03C01', '209281700007_R04C01', '209281700040_R03C01', '209281700123_R07C01', '209281700123_R05C01', '209281700107_R04C01', '209281700107_R03C01', '209281700107_R05C01', '209281700107_R02C01', '209281700107_R01C01', '209281700035_R03C01', '209281700123_R08C01', '209281700035_R02C01', '209281710051_R03C01', '209281700107_R08C01', '209281710051_R04C01', '209281700022_R08C01', '209281700123_R06C01', '209281700041_R01C01', '209281700041_R02C01', '209281700041_R05C01', '209281700041_R06C01', '209281700041_R07C01', '209281700123_R04C01', '209281700107_R07C01', '209281700107_R06C01', '209281700035_R01C01', '209281700106_R04C01', '209281700123_R03C01', '209281700123_R02C01', '209281710005_R08C01', '209281710005_R07C01', '209281710005_R06C01', '209281710005_R05C01', '209281710005_R04C01', '209281710005_R03C01', '209281710005_R02C01', '209281710005_R01C01', '209281710051_R02C01', '209281700106_R03C01', '209281700106_R05C01', '209281700106_R06C01', '209281700106_R07C01', '209281700106_R08C01', '209281710135_R01C01', '209281710051_R08C01', '209281710051_R07C01', '209281710051_R06C01', '209281710051_R05C01', '209281700106_R02C01']\n"
     ]
    }
   ],
   "source": [
    "filename_list = list(key['IID'])\n",
    "print(len(filename_list))\n",
    "print(filename_list)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Update the count with the last job # that you ran:\n",
    "count = 10\n",
    "\n",
    "# Define the number of codes to process per job (8)(Up this amount x3)\n",
    "codes_per_job = 60\n",
    "\n",
    "# Split the filename_list into chunks\n",
    "from itertools import zip_longest\n",
    "\n",
    "# Helper function to chunk the list into groups of 8\n",
    "def chunk_list(iterable, n):\n",
    "    args = [iter(iterable)] * n\n",
    "    return zip_longest(*args)\n",
    "\n",
    "# Loop through each chunk of 8 codes and create a single job\n",
    "for chunk in chunk_list(filename_list, codes_per_job):\n",
    "    # Filter out any None values from the last incomplete chunk\n",
    "    codes = [code for code in chunk if code is not None]\n",
    "    \n",
    "    # Generate a unique job name\n",
    "    count += 1\n",
    "    job_name = f'pedtobed{study.lower()}{count}'\n",
    "    \n",
    "    # Create a combined script for the N codes\n",
    "    script = f\"\"\"\n",
    "        #!/bin/bash\n",
    "\n",
    "        # Install and activate PLINK\n",
    "        chmod +x ./gp2_genotools_data/batch_files/plink_module.sh\n",
    "        ./gp2_genotools_data/batch_files/plink_module.sh\n",
    "    \"\"\"\n",
    "\n",
    "    # Add commands for each code in the chunk\n",
    "    for code in codes:\n",
    "        script += f\"\"\"\n",
    "        # Make analysis script executable and run for a specific code\n",
    "        chmod +x ./gp2_genotools_data/batch_files/convert_ped_to_bed.sh\n",
    "        ./gp2_genotools_data/batch_files/convert_ped_to_bed.sh {code}\n",
    "        \"\"\"\n",
    "\n",
    "    # Create the job with the combined script\n",
    "    create_script_job_with_bucket(\n",
    "        project_id=\"gp2-release-terra\",\n",
    "        region='europe-west4',\n",
    "        job_name=job_name,\n",
    "        bucket_name='gp2_genotools_data',\n",
    "        script_text=script\n",
    "    )\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "VETPD\n",
      "last job id 14\n",
      "Job Status: SUCCEEDED\n"
     ]
    }
   ],
   "source": [
    "# Initialize the client\n",
    "client = batch_v1.BatchServiceClient()\n",
    "job_id = f'pedtobed{study.lower()}{count}'\n",
    "\n",
    "# Define the job name (replace with your project ID, location, and job ID)\n",
    "project_id = \"gp2-release-terra\"\n",
    "region = 'europe-west4'\n",
    "job_name = f\"projects/{project_id}/locations/{region}/jobs/{job_id}\"\n",
    "\n",
    "# Get the job details\n",
    "job = client.get_job(name=job_name)\n",
    "\n",
    "# Check the status\n",
    "job_status = job.status.state\n",
    "print(study)\n",
    "print('last job id', count)\n",
    "print(f\"Job Status: {batch_v1.JobStatus.State(job_status).name}\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "There is 0 missing bed files for VETPD.\n",
      "[]\n"
     ]
    }
   ],
   "source": [
    "missing_beds = []\n",
    "\n",
    "for cohort in study_list:\n",
    "    df = key[key['study'] == cohort]\n",
    "    missing_cnt = 0\n",
    "\n",
    "    with open(f\"{raw_plink_path}/merge_bed_{cohort}.list\", 'w') as f:\n",
    "        for filename in df.IID:\n",
    "\n",
    "            bed = f'{raw_plink_path}/{filename}'\n",
    "            if os.path.isfile(f'{bed}.bed'):\n",
    "                f.write(f'{bed}\\n')\n",
    "            else:\n",
    "                #print(f'{bed} does not exist!!!')\n",
    "                missing_cnt += 1\n",
    "                missing_beds.append(filename)\n",
    "    f.close()\n",
    "    \n",
    "    print(f\"There is {missing_cnt} missing bed files for {cohort}.\")\n",
    "\n",
    "    #create missing_beds file for reference\n",
    "    with open(f'{missing_idat_dir}/missing_beds_{cohort}.txt', 'w') as f:\n",
    "        for m_bed in missing_beds:\n",
    "            f.write(f'{m_bed}\\n')\n",
    "    f.close()\n",
    "\n",
    "print(missing_beds)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Merge files"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>GP2sampleID_old</th>\n",
       "      <th>GP2sampleID</th>\n",
       "      <th>SentrixBarcode_A</th>\n",
       "      <th>SentrixPosition_A</th>\n",
       "      <th>study_old</th>\n",
       "      <th>study</th>\n",
       "      <th>sample_id</th>\n",
       "      <th>clinical_id</th>\n",
       "      <th>study_arm</th>\n",
       "      <th>study_type</th>\n",
       "      <th>...</th>\n",
       "      <th>family_history_other</th>\n",
       "      <th>family_history_pd</th>\n",
       "      <th>family_index</th>\n",
       "      <th>family_index_relationship</th>\n",
       "      <th>filename</th>\n",
       "      <th>FID</th>\n",
       "      <th>IID</th>\n",
       "      <th>pheno</th>\n",
       "      <th>sex_for_qc</th>\n",
       "      <th>GDPR</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>91592</th>\n",
       "      <td>NaN</td>\n",
       "      <td>VETPD_000193_s1</td>\n",
       "      <td>209281700115</td>\n",
       "      <td>R01C01</td>\n",
       "      <td>NaN</td>\n",
       "      <td>VETPD</td>\n",
       "      <td>GP2Z0061.1</td>\n",
       "      <td>GP2Z0061</td>\n",
       "      <td>Case</td>\n",
       "      <td>Case(/Control)</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>209281700115_R01C01</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>91593</th>\n",
       "      <td>NaN</td>\n",
       "      <td>VETPD_000194_s1</td>\n",
       "      <td>209281700115</td>\n",
       "      <td>R02C01</td>\n",
       "      <td>NaN</td>\n",
       "      <td>VETPD</td>\n",
       "      <td>GP2Z0062.1</td>\n",
       "      <td>GP2Z0062</td>\n",
       "      <td>Case</td>\n",
       "      <td>Case(/Control)</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>209281700115_R02C01</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>91594</th>\n",
       "      <td>NaN</td>\n",
       "      <td>VETPD_000195_s1</td>\n",
       "      <td>209281700115</td>\n",
       "      <td>R03C01</td>\n",
       "      <td>NaN</td>\n",
       "      <td>VETPD</td>\n",
       "      <td>GP2Z0063.1</td>\n",
       "      <td>GP2Z0063</td>\n",
       "      <td>Case</td>\n",
       "      <td>Case(/Control)</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>209281700115_R03C01</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>91595</th>\n",
       "      <td>NaN</td>\n",
       "      <td>VETPD_000196_s1</td>\n",
       "      <td>209281700115</td>\n",
       "      <td>R04C01</td>\n",
       "      <td>NaN</td>\n",
       "      <td>VETPD</td>\n",
       "      <td>GP2Z0064.1</td>\n",
       "      <td>GP2Z0064</td>\n",
       "      <td>Case</td>\n",
       "      <td>Case(/Control)</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>209281700115_R04C01</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>91596</th>\n",
       "      <td>NaN</td>\n",
       "      <td>VETPD_000197_s1</td>\n",
       "      <td>209281700007</td>\n",
       "      <td>R06C01</td>\n",
       "      <td>NaN</td>\n",
       "      <td>VETPD</td>\n",
       "      <td>GP2Z0068.1</td>\n",
       "      <td>GP2Z0068</td>\n",
       "      <td>Case</td>\n",
       "      <td>Case(/Control)</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>209281700007_R06C01</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>91779</th>\n",
       "      <td>NaN</td>\n",
       "      <td>VETPD_000380_s1</td>\n",
       "      <td>209281710051</td>\n",
       "      <td>R08C01</td>\n",
       "      <td>NaN</td>\n",
       "      <td>VETPD</td>\n",
       "      <td>GP2Z0731.1</td>\n",
       "      <td>GP2Z0731</td>\n",
       "      <td>Case</td>\n",
       "      <td>Case(/Control)</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>209281710051_R08C01</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>91780</th>\n",
       "      <td>NaN</td>\n",
       "      <td>VETPD_000381_s1</td>\n",
       "      <td>209281710051</td>\n",
       "      <td>R07C01</td>\n",
       "      <td>NaN</td>\n",
       "      <td>VETPD</td>\n",
       "      <td>GP2Z0732.1</td>\n",
       "      <td>GP2Z0732</td>\n",
       "      <td>Case</td>\n",
       "      <td>Case(/Control)</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>209281710051_R07C01</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>91781</th>\n",
       "      <td>NaN</td>\n",
       "      <td>VETPD_000382_s1</td>\n",
       "      <td>209281710051</td>\n",
       "      <td>R06C01</td>\n",
       "      <td>NaN</td>\n",
       "      <td>VETPD</td>\n",
       "      <td>GP2Z0733.1</td>\n",
       "      <td>GP2Z0733</td>\n",
       "      <td>Case</td>\n",
       "      <td>Case(/Control)</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>209281710051_R06C01</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>91782</th>\n",
       "      <td>NaN</td>\n",
       "      <td>VETPD_000383_s1</td>\n",
       "      <td>209281710051</td>\n",
       "      <td>R05C01</td>\n",
       "      <td>NaN</td>\n",
       "      <td>VETPD</td>\n",
       "      <td>GP2Z0734.1</td>\n",
       "      <td>GP2Z0734</td>\n",
       "      <td>Case</td>\n",
       "      <td>Case(/Control)</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>209281710051_R05C01</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>91783</th>\n",
       "      <td>NaN</td>\n",
       "      <td>VETPD_000384_s1</td>\n",
       "      <td>209281700106</td>\n",
       "      <td>R02C01</td>\n",
       "      <td>NaN</td>\n",
       "      <td>VETPD</td>\n",
       "      <td>GP2Z0735.1</td>\n",
       "      <td>GP2Z0735</td>\n",
       "      <td>Case</td>\n",
       "      <td>Case(/Control)</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>209281700106_R02C01</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>192 rows Ã— 51 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      GP2sampleID_old      GP2sampleID SentrixBarcode_A SentrixPosition_A  \\\n",
       "91592             NaN  VETPD_000193_s1     209281700115            R01C01   \n",
       "91593             NaN  VETPD_000194_s1     209281700115            R02C01   \n",
       "91594             NaN  VETPD_000195_s1     209281700115            R03C01   \n",
       "91595             NaN  VETPD_000196_s1     209281700115            R04C01   \n",
       "91596             NaN  VETPD_000197_s1     209281700007            R06C01   \n",
       "...               ...              ...              ...               ...   \n",
       "91779             NaN  VETPD_000380_s1     209281710051            R08C01   \n",
       "91780             NaN  VETPD_000381_s1     209281710051            R07C01   \n",
       "91781             NaN  VETPD_000382_s1     209281710051            R06C01   \n",
       "91782             NaN  VETPD_000383_s1     209281710051            R05C01   \n",
       "91783             NaN  VETPD_000384_s1     209281700106            R02C01   \n",
       "\n",
       "      study_old  study   sample_id clinical_id study_arm      study_type  ...  \\\n",
       "91592       NaN  VETPD  GP2Z0061.1    GP2Z0061      Case  Case(/Control)  ...   \n",
       "91593       NaN  VETPD  GP2Z0062.1    GP2Z0062      Case  Case(/Control)  ...   \n",
       "91594       NaN  VETPD  GP2Z0063.1    GP2Z0063      Case  Case(/Control)  ...   \n",
       "91595       NaN  VETPD  GP2Z0064.1    GP2Z0064      Case  Case(/Control)  ...   \n",
       "91596       NaN  VETPD  GP2Z0068.1    GP2Z0068      Case  Case(/Control)  ...   \n",
       "...         ...    ...         ...         ...       ...             ...  ...   \n",
       "91779       NaN  VETPD  GP2Z0731.1    GP2Z0731      Case  Case(/Control)  ...   \n",
       "91780       NaN  VETPD  GP2Z0732.1    GP2Z0732      Case  Case(/Control)  ...   \n",
       "91781       NaN  VETPD  GP2Z0733.1    GP2Z0733      Case  Case(/Control)  ...   \n",
       "91782       NaN  VETPD  GP2Z0734.1    GP2Z0734      Case  Case(/Control)  ...   \n",
       "91783       NaN  VETPD  GP2Z0735.1    GP2Z0735      Case  Case(/Control)  ...   \n",
       "\n",
       "      family_history_other family_history_pd family_index  \\\n",
       "91592                  NaN               NaN          NaN   \n",
       "91593                  NaN               NaN          NaN   \n",
       "91594                  NaN               NaN          NaN   \n",
       "91595                  NaN               NaN          NaN   \n",
       "91596                  NaN               NaN          NaN   \n",
       "...                    ...               ...          ...   \n",
       "91779                  NaN               NaN          NaN   \n",
       "91780                  NaN               NaN          NaN   \n",
       "91781                  NaN               NaN          NaN   \n",
       "91782                  NaN               NaN          NaN   \n",
       "91783                  NaN               NaN          NaN   \n",
       "\n",
       "      family_index_relationship filename FID                  IID pheno  \\\n",
       "91592                       NaN      NaN   0  209281700115_R01C01     2   \n",
       "91593                       NaN      NaN   0  209281700115_R02C01     2   \n",
       "91594                       NaN      NaN   0  209281700115_R03C01     2   \n",
       "91595                       NaN      NaN   0  209281700115_R04C01     2   \n",
       "91596                       NaN      NaN   0  209281700007_R06C01     2   \n",
       "...                         ...      ...  ..                  ...   ...   \n",
       "91779                       NaN      NaN   0  209281710051_R08C01     2   \n",
       "91780                       NaN      NaN   0  209281710051_R07C01     2   \n",
       "91781                       NaN      NaN   0  209281710051_R06C01     2   \n",
       "91782                       NaN      NaN   0  209281710051_R05C01     2   \n",
       "91783                       NaN      NaN   0  209281700106_R02C01     2   \n",
       "\n",
       "      sex_for_qc GDPR  \n",
       "91592          1    0  \n",
       "91593          1    0  \n",
       "91594          1    0  \n",
       "91595          1    0  \n",
       "91596          1    0  \n",
       "...          ...  ...  \n",
       "91779          1    0  \n",
       "91780          1    0  \n",
       "91781          1    0  \n",
       "91782          1    0  \n",
       "91783          1    0  \n",
       "\n",
       "[192 rows x 51 columns]"
      ]
     },
     "execution_count": 36,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "key"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "metadata": {},
   "outputs": [],
   "source": [
    "clin_key_dir = './gp2_genotools_data/clinical'\n",
    "for cohort in study_list:\n",
    "# write to files\n",
    "    df = key[key['study'] == cohort]\n",
    "    df[['FID','IID', 'FID', 'GP2sampleID']].to_csv(f'{clin_key_dir}/update_ids_{cohort}.txt', sep='\\t', header=False, index=False)\n",
    "    df[['FID', 'GP2sampleID', 'pheno']].to_csv(f'{clin_key_dir}/update_pheno_{cohort}.txt', sep='\\t', header=False, index=False)\n",
    "    df[['FID', 'GP2sampleID', 'sex_for_qc']].to_csv(f'{clin_key_dir}/update_sex_{cohort}.txt', sep='\\t', header=False, index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'./gp2_genotools_data/ped_bed'"
      ]
     },
     "execution_count": 38,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "raw_plink_path"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "metadata": {},
   "outputs": [],
   "source": [
    "missing_beds = []\n",
    "for cohort in study_list:\n",
    "    df = key[key['study'] == cohort]\n",
    "    with open(f\"{raw_plink_path}/merge_bed_{cohort}.list\", 'w') as f:\n",
    "        for filename in df.IID:\n",
    "            bed = f'{raw_plink_path}/{filename}'\n",
    "            if os.path.isfile(f'{bed}.bed'):\n",
    "                f.write(f'{bed}\\n')\n",
    "            else:\n",
    "                print(f'{bed} does not exist!!!')\n",
    "                missing_beds.append(filename)\n",
    "    f.close()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "./gp2_genotools_data/ped_bed/209281700115_R01C01\n",
      "./gp2_genotools_data/ped_bed/209281700115_R02C01\n",
      "./gp2_genotools_data/ped_bed/209281700115_R03C01\n",
      "./gp2_genotools_data/ped_bed/209281700115_R04C01\n",
      "./gp2_genotools_data/ped_bed/209281700007_R06C01\n",
      "./gp2_genotools_data/ped_bed/209281700115_R05C01\n",
      "./gp2_genotools_data/ped_bed/209281700007_R07C01\n",
      "./gp2_genotools_data/ped_bed/209281700007_R08C01\n",
      "./gp2_genotools_data/ped_bed/209281700040_R01C01\n",
      "./gp2_genotools_data/ped_bed/209281700115_R06C01\n"
     ]
    }
   ],
   "source": [
    "! head {raw_plink_path}/merge_bed_{cohort}.list"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'VETPD'"
      ]
     },
     "execution_count": 41,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "study"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "192 ./gp2_genotools_data/ped_bed/merge_bed_VETPD.list\n"
     ]
    }
   ],
   "source": [
    "! wc -l {raw_plink_path}/merge_bed_{cohort}.list"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 43,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Preparing bash script to merge bed files for VETPD.\n"
     ]
    }
   ],
   "source": [
    "# Create shell script to execute -- merge cohort files\n",
    "count = 0 # for Job ID count\n",
    "\n",
    "# Establish paths for inputs/outputs\n",
    "#out_path = f'./gp2_genotools_data/merged_by_cohort_r10/GP2_merge_{study}'\n",
    "out_path = f'./gp2_genotools_data/tests/GP2_merge_{study}_extra'\n",
    "\n",
    "plink_merge_cmd = f\"/home/levineks/bin/plink1.9/plink --merge-list {raw_plink_path}/merge_bed_{cohort}.list --update-ids {clin_key_dir}/update_ids_{cohort}.txt --make-bed --out {out_path}\"\n",
    "\n",
    "with open(f'./gp2_genotools_data/batch_files/merge_by_cohort.sh', 'w') as f:\n",
    "    f.write('#!/bin/bash\\n\\n')\n",
    "    f.write(f'{plink_merge_cmd}\\n')\n",
    "    f.close()\n",
    "\n",
    "print(f\"Preparing bash script to merge bed files for {study}.\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 44,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "#!/bin/bash\n",
      "\n",
      "/home/levineks/bin/plink1.9/plink --merge-list ./gp2_genotools_data/ped_bed/merge_bed_VETPD.list --update-ids ./gp2_genotools_data/clinical/update_ids_VETPD.txt --make-bed --out ./gp2_genotools_data/tests/GP2_merge_VETPD_extra\n"
     ]
    }
   ],
   "source": [
    "# Double check the shell script to make sure it does what you need and paths are correct\n",
    "! cat ./gp2_genotools_data/batch_files/merge_by_cohort.sh"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 45,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['VETPD']"
      ]
     },
     "execution_count": 45,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "study_list"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 46,
   "metadata": {},
   "outputs": [],
   "source": [
    "#Update the count with the last job # that you ran:\n",
    "count = 5\n",
    "\n",
    "# Set up the barcodes to process above\n",
    "\n",
    "# Loop through chromosomes and create separate jobs\n",
    "for study in study_list:\n",
    "    # Generate a unique job name\n",
    "    count += 1\n",
    "    job_name = f'mergebycohort{study.lower()}{count}'\n",
    "    \n",
    "    # Create a script for the specific chromosome\n",
    "    script = f\"\"\"\n",
    "        #!/bin/bash\n",
    "\n",
    "        # Install and activate PLINK2\n",
    "        chmod +x ./gp2_genotools_data/batch_files/plink_module.sh\n",
    "        ./gp2_genotools_data/batch_files/plink_module.sh\n",
    "        \n",
    "        # Make analysis script executable and run for a specific chromosome\n",
    "        chmod +x ./gp2_genotools_data/batch_files/merge_by_cohort.sh\n",
    "        ./gp2_genotools_data/batch_files/merge_by_cohort.sh\n",
    "    \"\"\"\n",
    "\n",
    "    # Create the job\n",
    "    create_script_job_with_bucket(\n",
    "        project_id=\"gp2-release-terra\",\n",
    "        region='europe-west4',\n",
    "        job_name=job_name,\n",
    "        bucket_name='gp2_genotools_data',\n",
    "        script_text=script\n",
    "    )"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 49,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "VETPD cohort merge\n",
      "Job Status: SUCCEEDED\n"
     ]
    }
   ],
   "source": [
    "# Initialize the client\n",
    "client = batch_v1.BatchServiceClient()\n",
    "job_id = f'mergebycohort{study.lower()}{count}'\n",
    "\n",
    "# Define the job name (replace with your project ID, location, and job ID)\n",
    "project_id = \"gp2-release-terra\"\n",
    "region = 'europe-west4'\n",
    "job_name = f\"projects/{project_id}/locations/{region}/jobs/{job_id}\"\n",
    "\n",
    "# Get the job details\n",
    "job = client.get_job(name=job_name)\n",
    "\n",
    "# Check the status\n",
    "job_status = job.status.state\n",
    "print(study, 'cohort merge')\n",
    "print(f\"Job Status: {batch_v1.JobStatus.State(job_status).name}\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 50,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "-rwxrwxrwx 1 levineks levineks  92M Feb 26 20:03 ./gp2_genotools_data/merged_by_cohort_r10/GP2_merge_VETPD.bed\n",
      "-rwxrwxrwx 1 levineks levineks  59M Feb 26 20:03 ./gp2_genotools_data/merged_by_cohort_r10/GP2_merge_VETPD.bim\n",
      "-rwxrwxrwx 1 levineks levineks 5.1K Feb 26 20:03 ./gp2_genotools_data/merged_by_cohort_r10/GP2_merge_VETPD.fam\n",
      "-rwxrwxrwx 1 levineks levineks 418K Feb 26 20:03 ./gp2_genotools_data/merged_by_cohort_r10/GP2_merge_VETPD.hh\n",
      "-rwxrwxrwx 1 levineks levineks 9.0M Feb 26 20:03 ./gp2_genotools_data/merged_by_cohort_r10/GP2_merge_VETPD.log\n",
      "-rwxrwxrwx 1 levineks levineks  110 Feb 26 20:03 ./gp2_genotools_data/merged_by_cohort_r10/GP2_merge_VETPD.nosex\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "-rwxrwxrwx 1 levineks levineks  92M Mar  7 14:59 ./gp2_genotools_data/tests/GP2_merge_VETPD_extra.bed\n",
      "-rwxrwxrwx 1 levineks levineks  59M Mar  7 14:59 ./gp2_genotools_data/tests/GP2_merge_VETPD_extra.bim\n",
      "-rwxrwxrwx 1 levineks levineks 5.1K Mar  7 14:59 ./gp2_genotools_data/tests/GP2_merge_VETPD_extra.fam\n",
      "-rwxrwxrwx 1 levineks levineks 620K Mar  7 14:59 ./gp2_genotools_data/tests/GP2_merge_VETPD_extra.hh\n",
      "-rwxrwxrwx 1 levineks levineks 9.0M Mar  7 14:59 ./gp2_genotools_data/tests/GP2_merge_VETPD_extra.log\n",
      "-rwxrwxrwx 1 levineks levineks   44 Mar  7 14:59 ./gp2_genotools_data/tests/GP2_merge_VETPD_extra.nosex\n"
     ]
    }
   ],
   "source": [
    "! ls -lh ./gp2_genotools_data/merged_by_cohort_r10/GP2_merge_{study}*\n",
    "! ls -lh ./gp2_genotools_data/tests/GP2_merge_{study}_extra*"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Merge old and new files together"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 51,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "VETPD\n",
      "382\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>GP2sampleID_old</th>\n",
       "      <th>GP2sampleID</th>\n",
       "      <th>SentrixBarcode_A</th>\n",
       "      <th>SentrixPosition_A</th>\n",
       "      <th>study_old</th>\n",
       "      <th>study</th>\n",
       "      <th>sample_id</th>\n",
       "      <th>clinical_id</th>\n",
       "      <th>study_arm</th>\n",
       "      <th>study_type</th>\n",
       "      <th>...</th>\n",
       "      <th>family_history_other</th>\n",
       "      <th>family_history_pd</th>\n",
       "      <th>family_index</th>\n",
       "      <th>family_index_relationship</th>\n",
       "      <th>filename</th>\n",
       "      <th>FID</th>\n",
       "      <th>IID</th>\n",
       "      <th>pheno</th>\n",
       "      <th>sex_for_qc</th>\n",
       "      <th>GDPR</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>91402</th>\n",
       "      <td>NaN</td>\n",
       "      <td>VETPD_000001_s1</td>\n",
       "      <td>209042130073</td>\n",
       "      <td>R02C01</td>\n",
       "      <td>NaN</td>\n",
       "      <td>VETPD</td>\n",
       "      <td>GP2Z0001.1</td>\n",
       "      <td>GP2Z0001</td>\n",
       "      <td>Case</td>\n",
       "      <td>Case(/Control)</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>209042130073_R02C01</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>91403</th>\n",
       "      <td>NaN</td>\n",
       "      <td>VETPD_000002_s1</td>\n",
       "      <td>209042130008</td>\n",
       "      <td>R06C01</td>\n",
       "      <td>NaN</td>\n",
       "      <td>VETPD</td>\n",
       "      <td>GP2Z0002.1</td>\n",
       "      <td>GP2Z0002</td>\n",
       "      <td>Case</td>\n",
       "      <td>Case(/Control)</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>209042130008_R06C01</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>91404</th>\n",
       "      <td>NaN</td>\n",
       "      <td>VETPD_000003_s1</td>\n",
       "      <td>209042130073</td>\n",
       "      <td>R07C01</td>\n",
       "      <td>NaN</td>\n",
       "      <td>VETPD</td>\n",
       "      <td>GP2Z0003.1</td>\n",
       "      <td>GP2Z0003</td>\n",
       "      <td>Case</td>\n",
       "      <td>Case(/Control)</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>209042130073_R07C01</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>91405</th>\n",
       "      <td>NaN</td>\n",
       "      <td>VETPD_000004_s1</td>\n",
       "      <td>209042130062</td>\n",
       "      <td>R06C01</td>\n",
       "      <td>NaN</td>\n",
       "      <td>VETPD</td>\n",
       "      <td>GP2Z0004.1</td>\n",
       "      <td>GP2Z0004</td>\n",
       "      <td>Case</td>\n",
       "      <td>Case(/Control)</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>209042130062_R06C01</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>91406</th>\n",
       "      <td>NaN</td>\n",
       "      <td>VETPD_000005_s1</td>\n",
       "      <td>209042130055</td>\n",
       "      <td>R04C01</td>\n",
       "      <td>NaN</td>\n",
       "      <td>VETPD</td>\n",
       "      <td>GP2Z0005.2</td>\n",
       "      <td>GP2Z0005</td>\n",
       "      <td>Case</td>\n",
       "      <td>Case(/Control)</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>209042130055_R04C01</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows Ã— 51 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      GP2sampleID_old      GP2sampleID SentrixBarcode_A SentrixPosition_A  \\\n",
       "91402             NaN  VETPD_000001_s1     209042130073            R02C01   \n",
       "91403             NaN  VETPD_000002_s1     209042130008            R06C01   \n",
       "91404             NaN  VETPD_000003_s1     209042130073            R07C01   \n",
       "91405             NaN  VETPD_000004_s1     209042130062            R06C01   \n",
       "91406             NaN  VETPD_000005_s1     209042130055            R04C01   \n",
       "\n",
       "      study_old  study   sample_id clinical_id study_arm      study_type  ...  \\\n",
       "91402       NaN  VETPD  GP2Z0001.1    GP2Z0001      Case  Case(/Control)  ...   \n",
       "91403       NaN  VETPD  GP2Z0002.1    GP2Z0002      Case  Case(/Control)  ...   \n",
       "91404       NaN  VETPD  GP2Z0003.1    GP2Z0003      Case  Case(/Control)  ...   \n",
       "91405       NaN  VETPD  GP2Z0004.1    GP2Z0004      Case  Case(/Control)  ...   \n",
       "91406       NaN  VETPD  GP2Z0005.2    GP2Z0005      Case  Case(/Control)  ...   \n",
       "\n",
       "      family_history_other family_history_pd family_index  \\\n",
       "91402                  NaN               NaN          NaN   \n",
       "91403                  NaN               NaN          NaN   \n",
       "91404                  NaN               NaN          NaN   \n",
       "91405                  NaN               NaN          NaN   \n",
       "91406                  NaN               NaN          NaN   \n",
       "\n",
       "      family_index_relationship filename FID                  IID pheno  \\\n",
       "91402                       NaN      NaN   0  209042130073_R02C01     2   \n",
       "91403                       NaN      NaN   0  209042130008_R06C01     2   \n",
       "91404                       NaN      NaN   0  209042130073_R07C01     2   \n",
       "91405                       NaN      NaN   0  209042130062_R06C01     2   \n",
       "91406                       NaN      NaN   0  209042130055_R04C01     2   \n",
       "\n",
       "      sex_for_qc GDPR  \n",
       "91402          1    0  \n",
       "91403          1    0  \n",
       "91404          1    0  \n",
       "91405          1    0  \n",
       "91406          1    0  \n",
       "\n",
       "[5 rows x 51 columns]"
      ]
     },
     "execution_count": 51,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# get key for entire study\n",
    "key = pd.read_csv(f'/home/levineks/gp2_genotools_data/clinical/GP2_master_key_MARCH_2025.txt', sep = '\\t', low_memory = False)\n",
    "key = key[key['study'] == study]\n",
    "print(study)\n",
    "print(len(key))\n",
    "key.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 52,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Move the old files to test folder/we'll overwrite in r10 folder with the new ones\n",
    "! cp /home/levineks/gp2_genotools_data/merged_by_cohort_r10/GP2_merge_{study}.* /home/levineks/gp2_genotools_data/tests/"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 53,
   "metadata": {},
   "outputs": [],
   "source": [
    "# New update IDs\n",
    "clin_key_dir = '/home/levineks/gp2_genotools_data/clinical'\n",
    "key[['FID','IID', 'FID', 'GP2sampleID']].to_csv(f'{clin_key_dir}/update_ids_{study}.txt', sep='\\t', header=False, index=False)\n",
    "key[['FID', 'GP2sampleID', 'pheno']].to_csv(f'{clin_key_dir}/update_pheno_{study}.txt', sep='\\t', header=False, index=False)\n",
    "key[['FID', 'GP2sampleID', 'sex_for_qc']].to_csv(f'{clin_key_dir}/update_sex_{study}.txt', sep='\\t', header=False, index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 54,
   "metadata": {},
   "outputs": [],
   "source": [
    "raw_plink_path = '/home/levineks/gp2_genotools_data/merged'\n",
    "\n",
    "missing_beds = []\n",
    "with open(f\"{raw_plink_path}/merge_bed_{study}.list\", 'w') as f:\n",
    "\n",
    "    bed1 = f'/home/levineks/gp2_genotools_data/tests/GP2_merge_{study}_extra'\n",
    "    if os.path.isfile(f'{bed1}.bed'):\n",
    "        f.write(f'{bed1}\\n')\n",
    "    else:\n",
    "        print(f'{bed1} does not exist!!!')\n",
    "\n",
    "    bed2 = f'/home/levineks/gp2_genotools_data/tests/GP2_merge_{study}'\n",
    "    if os.path.isfile(f'{bed2}.bed'):\n",
    "        f.write(f'{bed2}\\n')\n",
    "    else:\n",
    "        print(f'{bed2} does not exist!!!')    \n",
    "   \n",
    "f.close()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 55,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/home/levineks/gp2_genotools_data/tests/GP2_merge_VETPD_extra\n",
      "/home/levineks/gp2_genotools_data/tests/GP2_merge_VETPD\n"
     ]
    }
   ],
   "source": [
    "! cat {raw_plink_path}/merge_bed_{study}.list "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 56,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "382\n"
     ]
    }
   ],
   "source": [
    "! cat {clin_key_dir}/update_ids_{study}.txt | wc -l"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 57,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "192\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "190\n"
     ]
    }
   ],
   "source": [
    "! cat /home/levineks/gp2_genotools_data/tests/GP2_merge_{study}_extra.fam | wc -l\n",
    "! cat /home/levineks/gp2_genotools_data/tests/GP2_merge_{study}.fam | wc -l"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 58,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v1.90b7 64-bit (16 Jan 2023)             www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2023 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/levineks/gp2_genotools_data/merged_by_cohort_r10/GP2_merge_VETPD.log.\n",
      "Options in effect:\n",
      "  --make-bed\n",
      "  --merge-list /home/levineks/gp2_genotools_data/merged/merge_bed_VETPD.list\n",
      "  --out /home/levineks/gp2_genotools_data/merged_by_cohort_r10/GP2_merge_VETPD\n",
      "  --update-ids /home/levineks/gp2_genotools_data/clinical/update_ids_VETPD.txt\n",
      "\n",
      "32169 MB RAM detected; reserving 16084 MB for main workspace.\n",
      "Warning: Unusually long variant ID(s) present.  PLINK 1.9 does not scale well\n",
      "to length-80+ variant IDs; consider using a different naming scheme for long\n",
      "indels and the like.\n",
      "Warning: Variants 'ilmnseq_rs9701872_ilmndup1' and 'ilmnseq_rs9701872' have the\n",
      "same position.\n",
      "Warning: Variants 'exm2268640' and 'JHU_1.762319' have the same position.\n",
      "Warning: Variants 'rs35660652' and 'JHU_1.821925' have the same position.\n",
      "108706 more same-position warnings: see log file.\n",
      "Performing single-pass merge (382 people, 2004347 variants).\n",
      "Merged fileset written to                     \n",
      "/home/levineks/gp2_genotools_data/merged_by_cohort_r10/GP2_merge_VETPD-merge.bed\n",
      "+\n",
      "/home/levineks/gp2_genotools_data/merged_by_cohort_r10/GP2_merge_VETPD-merge.bim\n",
      "+\n",
      "/home/levineks/gp2_genotools_data/merged_by_cohort_r10/GP2_merge_VETPD-merge.fam\n",
      ".\n",
      "2004347 variants loaded from .bim file.\n",
      "382 people (362 males, 13 females, 7 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to\n",
      "/home/levineks/gp2_genotools_data/merged_by_cohort_r10/GP2_merge_VETPD.nosex .\n",
      "--update-ids: 0 people updated, 382 IDs not present.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 382 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Warning: 31800 het. haploid genotypes present (see\n",
      "/home/levineks/gp2_genotools_data/merged_by_cohort_r10/GP2_merge_VETPD.hh );\n",
      "many commands treat these as missing.\n",
      "Warning: Nonmissing nonmale Y chromosome genotype(s) present; many commands\n",
      "treat these as missing.\n",
      "Total genotyping rate is 0.981573.\n",
      "2004347 variants and 382 people pass filters and QC.\n",
      "Note: No phenotypes present.\n",
      "--make-bed to\n",
      "/home/levineks/gp2_genotools_data/merged_by_cohort_r10/GP2_merge_VETPD.bed +\n",
      "/home/levineks/gp2_genotools_data/merged_by_cohort_r10/GP2_merge_VETPD.bim +\n",
      "/home/levineks/gp2_genotools_data/merged_by_cohort_r10/GP2_merge_VETPD.fam ...\n",
      "101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n"
     ]
    }
   ],
   "source": [
    "#Merge files -- this should not take long because we're only merging 2 files\n",
    "merge_path = '/home/levineks/gp2_genotools_data/merged_by_cohort_r10'\n",
    "\n",
    "! /home/levineks/plink \\\n",
    "--merge-list {raw_plink_path}/merge_bed_{study}.list \\\n",
    "--update-ids {clin_key_dir}/update_ids_{study}.txt \\\n",
    "--make-bed \\\n",
    "--out {merge_path}/GP2_merge_{study}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 59,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "VETPD\n",
      "382\n"
     ]
    }
   ],
   "source": [
    "print(study)\n",
    "!cat /home/levineks/gp2_genotools_data/merged_by_cohort_r10/GP2_merge_{study}.fam | wc -l"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
